allocator.go

```diff
@@ -28,11 +26,19 @@ type Node struct {
 	resources map[Resource]int64
 }
 
-type Allocator struct {
+type options struct {
+	withNodeCapacity bool
+	withTagAffinity  bool
+}
+
+type OptionConfigurator func(*options)
```

Comment:
Edit: I guess this works, but it's definitely bad and an anti-pattern: https://stackoverflow.com/questions/37952354/golang-public-method-to-private-struct-is-this-have-any-use-case

These public methods on private types will not show up in your go docs. It'd be worthwhile to start adding comments and seeing how your docs would look like using the `godoc` tool.

Reply:
TIL, I made the change to prevent manual instantiation of the `allocator` struct, will revert.

Reply:
Re: `godoc`, I have a PR almost done for resource-balancing, I've been adding most of the comments there, since this PR is near completion, I'll hold off on adding those here and instead let them come up inside the next PR I'll be posting.